docs/bsp.md

```diff
@@ -370,20 +367,36 @@ where `MessageType` is defined as follows:
 object MessageType {
   /** An error message. */
   final val Error = 1
+
   /** A warning message. */
   final val Warning = 2
+
   /** An information message. */
   final val Info = 3
+
   /** A log message. */
   final val Log = 4
 }
 ```
 
+and a `TreeStructure` is defined as:
+
+```scala
+trait TreeStructure {
```

Comment:
I'm not so keen on this name. Given that `TreeStructure` is only used in two messages, can we revert this commit and keep `nodeId` and `parentId` inlined?

Reply:
If we keep it, this should be included in the "base data structures" section.